src/projects/EnsureThat/Enforcers/CollectionArg.cs

```diff
@@ -406,5 +406,312 @@ public T[] HasAny<T>([ValidatedNotNull, NotNull]T[] value, Func<T, bool> predica
 
             return value;
         }
+
+        [return: NotNull]
+        [ContractAnnotation("value:null => halt")]
+        public ICollection<T> HasAllNotNullItems<T>([ValidatedNotNull, NotNull]ICollection<T> value, [InvokerParameterName, CallerArgumentExpression(nameof(value))] string paramName = null, OptsFn optsFn = null) where T : class
+        {
+            Ensure.Any.IsNotNull(value, paramName);
+
+            Ensure.Collection.HasItems(value, paramName);
```

Comment:
why new line though?

Reply:
This method checks that:
1. Collection is not null
2. Collection has items
3. All items in collection are not null/default

Reply:
Is the ```IsNotNull``` check necessary? Since ```HasItems``` contains this check